### PR TITLE
1.18+ Book editing feature

### PR DESCRIPTION
**Pull request by unlimitedcoder2**
In 1.18 a new book editing packet was added
https://github.com/prismarinejs/mineflayer/issues/3204

### CODE REVIEW

File: data/pc/common/features.json [hunk 1]
```diff
@@ -238,9 +238,17 @@
...
     "name": "hasEditBookPacket",
-    "description": "book editing is handled through a packet",
+    "description": "book editing is handled through a packet with the updated book item",
     "versions": [
       "1.13",
+	  "1.17.1"
+    ]
+  },
+  {
+    "name": "hasNonNbtEditBookPacket",
```
Comment (extremeheat) [inline data/pc/common/features.json:248]: The feature naming is confusing here. Can you elaborate what exactly the difference in packet is before vs after?

Comment (unlimitedcoder2) [reply data/pc/common/features.json:248]: The old edit book packet serializes the item and uses the nbt data from that
image

New packet sends the pages and title as packet fields
image

Comment (extremeheat) [reply data/pc/common/features.json:248]: How about adding a `editBookPacketUsesNbt` feature then? This seems clearer than the current naming between `hasEditBookPacket` and `hasNonNbtEditBookPacket`, as there is actaully an edit book packet on both version ranges.

Comment (extremeheat) [reply data/pc/common/features.json]: Done

**Review by extremeheat — APPROVED** [review]

**Pull request merged**